packages/client/src/features/debug/debug-bounds-decorator.tsx

```diff
@@ -0,0 +1,173 @@
+/********************************************************************************
+ * Copyright (c) 2024 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+/* eslint-disable max-len */
+
+import { Bounds, GModelElement, IVNodePostprocessor, Point, isSizeable, setClass, svg } from '@eclipse-glsp/sprotty';
+import { inject, injectable, optional } from 'inversify';
+import { VNode } from 'snabbdom';
+import { GGraph } from '../../model';
+import { BoundsAwareModelElement } from '../../utils';
+import { DebugManager } from './debug-manager';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const JSX = { createElement: svg };
+
+@injectable()
+export class DebugBoundsDecorator implements IVNodePostprocessor {
+    @optional()
+    @inject(DebugManager)
+    protected debugManager?: DebugManager;
+
+    decorate(vnode: VNode, element: GModelElement): VNode {
+        if (!this.debugManager?.isDebugEnabled) {
+            return vnode;
+        }
+        if (isSizeable(element)) {
```

Comment:
I think I used this one because it checks the `boundsFeature` for the element and therefore a bit more restrictive as other elements (such as decorations) may also have bounds but not the feature. I can't say which one is better, maybe for a debugging case being less restrictive is actually better, I'll change it to the bounds aware check.